File: 11-load&initialize/11-load&initialize/ViewController.m
//
//  ViewController.m
//  11-load&initialize
//
//  Created by jackfrow on 2019/4/28.
//  Copyright © 2019 jackfrow. All rights reserved.
//

#import "ViewController.h"
#import "ClassB.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    ClassB* cb = [[ClassB alloc] init];
    NSLog(@"cb = %@",cb.description);
    
    ClassB* cb2 =[[ClassB alloc] init];
    NSLog(@"cb2 = %@",cb2.description);
    
    
    
}


@end

File: 05-ViewCrashBug/05-ViewCrashBug/ViewController.swift
//
//  ViewController.swift
//  05-ViewCrashBug
//
//  Created by jackfrow on 2018/12/28.
//  Copyright © 2018 jackfrow. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
       
      
        testJson()
        
    }
    
    
    func testJson()  {
        
       let str = "赵云"
        let data = str.data(using: .utf8)
        
        do {
            try  let json =  JSONSerialization.jsonObject(with: data!, options: .mutableContainers)
            
            
        } catch  {
            
            print("decode failed")
            
        }

        
        
    }
    
    


}

